setup.py

```diff
@@ -1022,11 +1022,19 @@ def main():
         'opt-einsum': ['opt-einsum>=3.3']
     }
     if platform.system() == 'Linux':
-        triton_pin_file = os.path.join(cwd, ".github", "ci_commit_pins", "triton.txt")
-        if os.path.exists(triton_pin_file):
-            with open(triton_pin_file) as f:
-                triton_pin = f.read().strip()
-                extras_require['dynamo'] = ['pytorch-triton==2.0.0+' + triton_pin[:10], 'jinja2']
+        if os.path.exists('/opt/rocm'):
+            # assume rocm build
+            triton_pin_file = os.path.join(cwd, ".github", "ci_commit_pins", "triton-rocm.txt")
+            if os.path.exists(triton_pin_file):
+                with open(triton_pin_file) as f:
+                    triton_pin = f.read().strip()
+                    extras_require['dynamo'] = ['pytorch-triton-rocm==2.0.0+' + triton_pin[:10], 'jinja2']
+        else:
+            triton_pin_file = os.path.join(cwd, ".github", "ci_commit_pins", "triton.txt")
+            if os.path.exists(triton_pin_file):
+                with open(triton_pin_file) as f:
+                    triton_pin = f.read().strip()
+                    extras_require['dynamo'] = ['pytorch-triton==2.0.0+' + triton_pin[:10], 'jinja2']
```

Comment:
Plz, avoid code duplication
```suggestion
       pin_name, pkg_name = ("pytorch-trition", "trition.txt") is not os.path.exists("/opt/rocm") else ("pytorch-trition-rocm", "trition-rocm.txt")
        triton_pin_file = os.path.join(cwd, ".github", "ci_commit_pins", pin_name)
        if os.path.exists(triton_pin_file):
            with open(triton_pin_file) as f:
                triton_pin = f.read().strip()
                extras_require['dynamo'] = [f'{pkg_name}==2.0.0+' + triton_pin[:10], 'jinja2']
```